src/repository/soldierRepo.ts

```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import logger from '../logger';
+import { getDatabase } from './mongoConnect';
+
+interface Soldier {
+  _id?:any;
+  id: string;
+  name: string;
+  rank: string;
+  limitations:Array<string>;
+  duties?:Array<string>;
+}
+
+export async function createSoldier(soldier: Soldier) {
+  try {
+    // eslint-disable-next-line no-underscore-dangle, no-param-reassign
+    soldier._id = +soldier.id;
+    await getDatabase().collection('soldiers').insertOne(soldier);
+    return soldier;
+  } catch (err) {
+    logger.info(err);
+    throw err;
+  }
+}
+export async function getSoldiers(searchId:string): Promise<any> {
```

Comment:
You don't need to write the return type of the function explicitly, and if you do, don't write Promise<any>

Reply:
Instead you can return the object you return `as <type>` or `satisfies <type>`.


And make this a Soldier